File: z_immortal/code/.cursor/hooks/mode2_disarm.py
#!/usr/bin/env python3
"""Disarm Mode-2 session marker."""
from __future__ import annotations

from pathlib import Path

ROOT = Path(__file__).resolve().parents[2]
MARKER = ROOT / ".cursor" / "mode2-session.json"


def main() -> None:
    if MARKER.exists():
        MARKER.unlink()
        print("MODE2_DISARMED")
    else:
        print("MODE2_ALREADY_OFF")


if __name__ == "__main__":
    main()

File: z_immortal/code/.cursor/hooks/mode2_disarm.js
#!/usr/bin/env node
const fs = require("fs");
const path = require("path");
const { writeStatus } = require("./mode2_status.js");

const ROOT = path.resolve(__dirname, "../..");
const MARKER = path.join(ROOT, ".cursor", "mode2-session.json");

if (fs.existsSync(MARKER)) {
  fs.unlinkSync(MARKER);
  writeStatus("paused", "Mode-2 paused. Send 2 to start again.");
  console.log("MODE2_DISARMED");
} else {
  writeStatus("idle", "No Mode-2 session.");
  console.log("MODE2_ALREADY_OFF");
}